File: Classes/OutputClasses/EngineClasses/Throttle.py
from ...GeneralClasses.ControlSurface import ControlSurface

class Throttle(ControlSurface):
    def __init__(self) -> None:
        super().__init__()
        self.min_position.set(0)
        self.change_rate.set(1)

    def deploy(self):
        if not self.move_to_target():
            return
            #print(f'\n{self.name.get()} is not in tolerance.')

    def retract(self):
        self.return_to_zero()

    def run(self):
        self.deploy()

File: C++ Implimentation/Classes/Engines/Engine/Throttle/Throttle.cpp
#include "Throttle.h"

    Throttle::Throttle() {
    min_position.setValue(0);
    change_rate.setValue(1);
}

void Throttle::deploy(){
    if (move_to_target().getNotValue()){
        return;
        //#print(f'\n{self.name.get()} is not in tolerance.')
        }
}

void Throttle::retract(){
    return_to_zero();
}

void Throttle::run(){
    deploy();
}
